### PR TITLE
Feature enable export reports to csv file

### PR DESCRIPTION
**Pull request by rwrzesien**
Resolves https://github.com/Code-Poets/sheetstorm/issues/311

### CODE REVIEW

**Comment by rwrzesien** [comment]
@dybi Yes, it is intended, as it is not technically possible for CSV file to have more than 1 sheet, so I have assumed that we want to export only single user report. If we want to export multi-sheet project report too, I would need a sample of output file structure. @Karrp do we want that and could you provide that sample?

**Comment by kbeker** [comment]
@rwrzesien I think that @Karrp is now unreachable so I think that structure of this output may be as follows:
```
archive named by project_name
project_name_creation_date.zip
| name_firstLetterOfLastName1.csv
| name_firstLetterOfLastName2.csv
.
.
```
I think that this may looks like this. What do you think guys @rwrzesien @dybi ?

**Comment by rwrzesien** [comment]
@kbeker Nice idea, I like it! Will require some extra work but definitely doable.

**Comment by rwrzesien** [comment]
@dybi @kbeker Updated and ready for review.

File: employees/common/exports.py [hunk 1]
```diff
@@ -216,3 +224,58 @@ def _get_report_date_and_daily_hours(self, current_report: Report, reports_subse
...
+                    total_hours += timezone.timedelta(hours=int(hours), minutes=int(minutes))
+            elif is_last_row and cell.data_type is TYPE_FORMULA:
+                next_row.append(str(total_hours)[:-3])
+            else:
+                next_row.append(cell.value)
+
+        writer.writerow(next_row)
+
+
+def save_work_book_as_zip_of_csv(work_book: Workbook) -> zipfile.ZipFile:
```
Comment (kbeker) [inline employees/common/exports.py]: Don't you return here `BytesIO` object?

File: employees/templates/employees/project_report_list.html [hunk 2]
```diff
@@ -23,7 +23,10 @@ <h1>
 </h1>
 {% with date=path|extract_year_and_month_from_url %}
 <a href="{% url 'export-project-data-xlsx' pk=object.pk year=date.0 month=date.1 %}" class="btn btn-info">
-        Export Reports
+    Export Reports
```
Comment (kbeker) [inline employees/templates/employees/project_report_list.html]: Please add here that this button gives `XLSX` export

File: employees/templates/employees/report_list.html [hunk 3]
```diff
@@ -100,6 +100,9 @@ <h4><font color="red">{{ error }}</font></h4>
             <a href="{% url 'export-data-xlsx' pk=user.pk year=date.0 month=date.1 %}" class="btn btn-info">
                 Export {{ date.1 | convert_to_month_name }} {{ date.0 }} Reports
```
Comment (kbeker) [inline employees/templates/employees/report_list.html]: Please add here that this gives `XLSX` export :)

File: employees/common/exports.py [hunk 4]
```diff
@@ -216,3 +224,58 @@ def _get_report_date_and_daily_hours(self, current_report: Report, reports_subse
...
+            continue
+
+        # Check if is last row by comparing first cell value to `TOTAL`.
+        if isinstance(row[0].value, str) and row[0].value == constants.TOTAL.value:
+            is_last_row = True
+
+        next_row = []
+        for cell_number, cell in enumerate(row, start=1):
+            if isinstance(cell.value, str) and cell.value.lower().startswith("=timevalue"):
+                hours_as_string = cell.value[len('=timevalue("') : -len('")')]
```
[the remark this answers is not in the document]
Comment (rwrzesien) [reply employees/common/exports.py]: This is being set by `black`, so we can either disable it `black` config, or disable it in `pylint` config, or disable it only for this single line, what do you think would be best in this case?

Btw. where is `black` config file in our repo? I couldn't find it.

File: employees/common/exports.py [hunk 5]
```diff
@@ -216,3 +224,58 @@ def _get_report_date_and_daily_hours(self, current_report: Report, reports_subse
 
 def convert_markdown_html_to_text(html: str) -> str:
     return "".join(BeautifulSoup(html, features="html.parser").findAll(text=True))
+
+
+def save_work_book_as_csv(writer: csv.DictWriter, work_book: Workbook) -> None:
+    sheet = work_book.get_active_sheet()
```
[the remark this answers is not in the document]
Comment (rwrzesien) [reply employees/common/exports.py]: :+1 done

File: employees/common/exports.py (hunk 1 above)
Comment (rwrzesien) [reply employees/common/exports.py]: Right, ZIP was returned in previous version of the code. :+1: done

File: employees/templates/employees/project_report_list.html (hunk 2 above)
Comment (rwrzesien) [reply employees/templates/employees/project_report_list.html]: :+1: done

File: employees/templates/employees/report_list.html (hunk 3 above)
Comment (rwrzesien) [reply employees/templates/employees/report_list.html]: :+1: done

File: employees/common/exports.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (kbeker) [reply employees/common/exports.py]: @rwrzesien Actually we don't have black config. We give parameters explicitly to `black` command. Can we by `black` config disable only one line in whole file?

**Pull request merged**